src/modules/Elsa.Alterations.Core/Services/WorkflowInstanceFinder.cs

```diff
@@ -0,0 +1,74 @@
+using Elsa.Alterations.Core.Contracts;
+using Elsa.Alterations.Core.Models;
+using Elsa.Workflows;
+using Elsa.Workflows.Management.Contracts;
+using Elsa.Workflows.Management.Filters;
+using Elsa.Workflows.Runtime.Contracts;
+using Elsa.Workflows.Runtime.Filters;
+
+namespace Elsa.Alterations.Core.Services;
+
+/// <inheritdoc />
+public class WorkflowInstanceFinder(IWorkflowInstanceStore workflowInstanceStore, IActivityExecutionStore activityExecutionStore) : IWorkflowInstanceFinder
+{
+    /// <inheritdoc />
+    public async Task<IEnumerable<string>> FindAsync(AlterationWorkflowInstanceFilter filter, CancellationToken cancellationToken = default)
+    {
+        var workflowInstanceFilter = new WorkflowInstanceFilter
+        {
+            Ids = filter.WorkflowInstanceIds?.ToList(),
+            DefinitionVersionIds = filter.DefinitionVersionIds?.ToList(),
+            CorrelationIds = filter.CorrelationIds?.ToList(),
+            HasIncidents = filter.HasIncidents,
+            IsSystem = filter.IsSystem,
+            TimestampFilters = filter.TimestampFilters?.ToList(),
+            WorkflowStatus = WorkflowStatus.Running
+        };
+        var activityExecutionFilters = filter.ActivityFilters?.Select(x => new ActivityExecutionRecordFilter
+        {
+            ActivityId = x.ActivityId,
+            Id = x.ActivityInstanceId,
+            ActivityNodeId = x.NodeId,
+            Name = x.Name,
+            Status = x.Status,
+        }).ToList();
+
+        var workflowInstanceFilterIsEmpty = WorkflowFilterIsEmpty(workflowInstanceFilter);
+
+        var workflowInstanceIds = workflowInstanceFilterIsEmpty
+            ? Enumerable.Empty<string>().ToHashSet()
```

Comment:
Do we give some feedback to the user that they actually need to enter filter criteria or is this implied from the empty list that is returned?